lib/ex_plasma/transaction/protocol.ex

```diff
@@ -0,0 +1,46 @@
+defprotocol ExPlasma.Transaction.Protocol do
```

Comment:
Maybe a behaviour should be used here?
quote from Jose Valim
```
A behaviour is a way to say: give me a module as an argument and I will invoke the following callbacks on it. A more complex example for behaviours besides a GenServer are the Ecto adapters. However, this does not work if you have a data structure and you want to dispatch based on the data structure. Hence protocols.
```
Also, I have a post about it in my blog https://www.badykov.com/ethereum/2018/12/29/elixir-behaviours-for-blockchain-hardfork-configuration/

Reply:
Hmm, but doesn't the Protocol fits better here as we do have different data structure?